clients/src/main/java/org/apache/kafka/clients/producer/Producer.java

```diff
@@ -33,19 +35,80 @@
  * @see MockProducer
  */
 public interface Producer<K, V> extends Closeable {
+    public static final int INVALID_PID = Integer.MAX_VALUE;
```

Comment:
Maybe we should use a negative number as the sentinel. It's a bit more conventional and it stands out much clearer as invalid in logging and such. There may not be a great place to put this at the moment, but we have tended to locate it with the Request/Response objects. Could also be located with `TransactionState`.